lib/undercover/plugin.rb

```diff
@@ -17,32 +17,60 @@ module Danger
   class DangerUndercover < Plugin
     VALID_FILE_FORMAT = '.txt'
     DEFAULT_PATH = 'coverage/undercover.txt'
+    MAX_INLINE_COMMENTS = 30
+    MAX_CHARACTERS_PER_MESSAGE = 60_000
 
-    # Checks the file validity and warns if no file is found
-    # if a valid file is found then if there are no changes,
-    # shows the report as a message in Danger.
-    # If there are reports then it shows the report as a warning in danger.
+    # Checks the file validity and fails if no file is found.
+    # If a valid file is found, the content of it will be parsed according to options
+    # and reports will be sent to danger.
+    #
+    # Options
+    #   - *undercover_path* the path to the undercover.txt file.
+    #     Defaults to `coverage/undercover.txt`.
+    #   - *sticky* from (danger)[https://danger.systems/reference.html].
+    #     Defaults to `true`.
+    #   - *in_line* when `true` will report each line or block missing coverage as an in-line comment.
+    #     Defaults to `false`.
+    #   - *report_danger* when `true` will report a failure to danger when new code is at 0 test coverage.
+    #     Defaults to `false`, so it generates warnings instead.
+    #   - *max_inline_comments* determine the maximum number of reported messages.
+    #     Used to avoid Github API abuse mechanism.
+    #     When limit is reached, a custom message is added to inform the user.
+    #     Defaults to 10.
+    #
     # @return  [void]
     #
-    def report(undercover_path = DEFAULT_PATH, sticky: true, in_line: false, fail_on_missing_coverage: false)
+    def report(undercover_path = DEFAULT_PATH, sticky: true, in_line: false, report_danger: false, max_inline_comments: MAX_INLINE_COMMENTS)
```

Comment:
It's unclear to me why `fail_on_missing_coverage` was renamed. I found it much more descriptive than `report_danger`

Reply:
Ah! I forgot to mention the why of this :) 
It is to use a common interface across Dangerfile plugins. 
We currently use already the `danger-rubocop` plugin which has a setting to achieve the same, and they call it `report_danger` https://github.com/ashfurrow/danger-rubocop#methods so I copied over :D

Reply:
It doesn't feel truly the same - it seems that there's also another way to fail CI in `danger-rubocop`, via `fail_on_inline_comment`:
https://github.com/ashfurrow/danger-rubocop/blob/4a5cf7cee9d206c035f6f736c24a77bf0641dd8a/lib/danger_plugin.rb#L138-L142

`report_danger` might have been a leftover name from early days and stayed for the compatibility. The `fail_on_missing_coverage` certainly feels more descriptive to me.

It's not a big deal if you like `report_danger` more though :)

Reply:
I'd consider dropping `max_inline_comments` as a parameter?  It's pretty much a "do you want this to break unexpectedly on a large PR" parameter, to which to obvious answer is surely always "no"?